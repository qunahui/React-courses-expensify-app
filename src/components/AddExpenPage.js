import React from 'react';
import ExpenseForm from './ExpenseForm';
import { connect } from 'react-redux';
import { addExpense } from '../actions/expenses'

export class AddExpenPage extends React.Component {
    onSubmit = (expense) => {
        this.props.addExpense(expense);
        this.props.history.push("/");
    }

    render() {
        return (
            <div>
                <h1>Add expense</h1>
                <ExpenseForm
                    onSubmit={this.onSubmit}
                />
            </div>
        )
    }
}

// const AddExpenPage = (props) => (
//     <div>
//         <h1>Add expense</h1>
//         <ExpenseForm
//             onSubmit={(expense) => {
//                 props.dispatch(addExpense(expense));
//                 props.history.push("/");
//             }}
//         />
//     </div>
// );

const mapDispatchToProps = (dispatch) => {
    return {
        addExpense: (expense) => dispatch(addExpense(expense))
    }
}

export default connect(undefined, mapDispatchToProps)(AddExpenPage);