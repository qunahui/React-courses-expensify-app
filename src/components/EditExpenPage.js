import React from 'react'
import { connect } from 'react-redux'
import ExpenseForm from './ExpenseForm';
import { startEditExpense, startRemoveExpense } from '../actions/expenses'

export class EditExpenPage extends React.Component {
    onSubmit = (expense) => {
        this.props.startEditExpense(this.props.expense.id, expense);
        this.props.history.push("/");
    }

    onClick = () => {
        this.props.startRemoveExpense({ id: this.props.expense.id });
        this.props.history.push("/");
    }

    render() {
        return (
            <div>
                <div class="page-header">
                    <div className="content-container">
                        <h1 className="page-header__title">Edit Expense</h1>
                    </div>
                </div>
                <div className="content-container">
                    <ExpenseForm
                        expense={this.props.expense}
                        onSubmit={this.onSubmit}
                    />
                    <button
                        className="button button--secondary"
                        onClick={this.onClick}>
                        Remove Expense</button>
                </div>
            </div>
        )
    }
}

const mapStateToProps = (state, props) => {
    return {
        expense: state.expenses.find((expense) => {
            return expense.id === props.match.params.id;
        })
    };
}

const mapDispatchToProps = (dispatch) => {
    return {
        startEditExpense: (id, expenseEdit) => dispatch(startEditExpense(id, expenseEdit)),
        startRemoveExpense: (data) => dispatch(startRemoveExpense(data))
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(EditExpenPage);