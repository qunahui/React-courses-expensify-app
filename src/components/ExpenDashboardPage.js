import React from 'react'
import ExpenseList from './ExpenseList'
import ExpenseListFilters from './ExpenseListFilters'
import ExpenseSummary from './ExpenseSummary'

const ExpenDashboardPage = () => (
    <div>
        <ExpenseSummary />
        <ExpenseListFilters />
        <ExpenseList />
    </div>
)

export default ExpenDashboardPage;