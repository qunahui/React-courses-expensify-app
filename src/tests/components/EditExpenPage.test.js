import React from 'react'
import { shallow } from 'enzyme'
import { EditExpenPage } from '../../components/EditExpenPage'
import expenses from '../fixtures/expenses'

let editExpense, removeExpense, history, wrapper

beforeEach(() => {
    editExpense = jest.fn();
    removeExpense = jest.fn();
    history = { push: jest.fn() };
    wrapper = shallow(
        <EditExpenPage
            expense={expenses[1]}
            editExpense={editExpense}
            removeExpense={removeExpense}
            history={history}
        />
    );
});

test('should render EditExpenPage', () => {
    expect(wrapper).toMatchSnapshot();
})

test('should handle editExpenPage', () => {
    const edit = {
        note: 'this is edit note'
    }
    wrapper.find('ExpenseForm').prop('onSubmit')(edit);
    expect(history.push).toHaveBeenCalledWith('/');
    expect(editExpense).toHaveBeenLastCalledWith(expenses[1].id, edit);
})
