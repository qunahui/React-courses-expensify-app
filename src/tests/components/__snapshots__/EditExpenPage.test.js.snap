// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render EditExpenPage 1`] = `
<div>
  <ExpenseForm
    expense={
      Object {
        "amount": 195000,
        "createAt": -345600000,
        "description": "Rent",
        "id": "1",
        "note": "",
      }
    }
    onSubmit={[Function]}
  />
  <button
    onClick={[Function]}
  >
    Remove
  </button>
</div>
`;
